### DISCUSSION TITLE
Field Array Dirty Fields

### DISCUSSION THREAD

**Discussion by Daneng66** [react-hook-form/react-hook-form#7449] [Q&A]
When using field arrays the dirty field state returns an object containing all the fields in the array and their dirty state, is there a way just to have a single value if any of the fields in the array has changed?

**Comment by bluebill1049** [comment]
it's return fields with boolean true and false.

**Reply by bluebill1049** [reply]
It's the fastest way for the lib to return dirty fields which return boolean true/false, if you want to only have true value, you can write a function at your app level to filter them out.

**Reply by Daneng66** [reply]
Its not the case I only want true values, I only want a single true/false if any fields have changed

**Comment by Moshyfawn** [comment]
[Form state has `dirtyFields` object](https://react-hook-form.com/api/useform/formstate), that's populated with user-modified fields. You can check if it contains an object with your field of interest by just accessing it `dirtyFields.fieldOfInterest` and converting it into a `boolean`. See [this codesandbox for an example](https://codesandbox.io/s/has-dirty-fields-3wzns).
Make sure to provide `defaultValues` for the form to compare against the current input values to determine if the field has been modified or not.
